src/Chess.cpp

```diff
@@ -0,0 +1,257 @@
+bool Chess::isSame() const 
+{
+	return ((m_input[0] == m_input[2]) && (m_input[1] == m_input[3]));
+} 
+// check if the input is lockations at board
+bool Chess::isValid() const
+{
+	return ((('A' <= m_input[0]) && (m_input[0] <= 'H')) || (('a' <= m_input[0]) && (m_input[0] <= 'h')) &&
+		(('1' <= m_input[1]) && (m_input[1] <= '8')) &&
+		(('A' <= m_input[2]) && (m_input[2] <= 'H')) || (('a' <= m_input[2]) && (m_input[2] <= 'h')) &&
+		(('1' <= m_input[3]) && (m_input[3] <= '8')));
+}
+	
+// check if the input is exit or quit  
+bool Chess::isExit() const 
+{
+	return ((m_input == "exit") || (m_input == "quit") || (m_input == "EXIT") || (m_input == "QUIT"));
+}
+// execute the movement on board 
+void Chess::excute()
+{
+	int row = (m_input[0] - 'a');
+	int col = (m_input[1] - '1');
+	char pieceInSource = m_boardString[(row * 8) + col]; 
+	m_boardString[(row * 8) + col] = '#'; 
+
+	row = (m_input[2] - 'a');
+	col = (m_input[3] - '1');
+	m_boardString[(row * 8) + col] = pieceInSource; 
+
+	setPieces(); 
+}
+// check the response code and switch turn if needed 
+void Chess::doTurn()
+{
+	m_errorMsg = "\n"; 
+	switch (m_codeResponse)
+	{
+	case 11:
+	{
+		m_msg = "there is not piece at the source \n";
+		break;
+	}
+	case 12:
+	{
+		m_msg = "the piece in the source is piece of your opponent \n";
+		break;
+	}
+	case 13:
+	{
+		m_msg = "there one of your pieces at the destination \n";
+		break;
+	}
+	case 21:
+	{
+		m_msg = "illegal movement of that piece \n";
+		break;
+	}
+	case 31:
+	{
+		m_msg = "this movement will cause you checkmate \n";
+		break;
+	}
+	case 41:
+	{
+		excute();
+		m_turn = !m_turn;
+		m_msg = "the last movement was legal and cause check \n";
+		break;
+	}
+	case 42:
+	{
+		excute();
+		m_turn = !m_turn;
+		m_msg = "the last movement was legal \n";
+		break;
+	}
+	}
+}
+
+// C'tor
+Chess::Chess(const string& start)
+	: m_boardString(start),m_codeResponse(-1)
+{
+	setFrames();
+	setPieces();
+}
+
+// get the source and destination 
+string Chess::getInput()
+{
+	static bool isFirst = true;
+
+	if (isFirst)
+		isFirst = false;
+	else
+		doTurn(); 
+
+	displayBoard();
+	showAskInput();
+
+	cin >> m_input;
+	if (isExit())
+		return "exit";
+	while (!isValid() || isSame())
+	{
+		if (!isValid())
+			m_errorMsg = "Invalid input !! \n";
+		else
+			m_errorMsg = "The source and the destination are the same !! \n";
+		displayBoard();
+		showAskInput();
+		cin >> m_input;
+		if (isExit())
+			return "exit";
+	}
+
+	if (m_input != "exit")
+	{
+		if (('A' <= m_input[0]) && (m_input[0] <= 'H'))
+			m_input[0] = (m_input[0] - 'A' + 'a');
+		if (('A' <= m_input[2]) && (m_input[2] <= 'H'))
+			m_input[2] = (m_input[2] - 'A' + 'a');
+	}
+
+	return m_input;
+}
+
+void Chess::setCodeResponse(int codeResponse)
+{
+	if (((11 <= codeResponse) && (codeResponse <= 13)) ||
+		((21 == codeResponse) || (codeResponse == 31)) ||
+		((41 == codeResponse) || (codeResponse == 42)))
+		m_codeResponse = codeResponse;

```

Comment:
Unnecessary test It is possible in the switch case to do the Default case which is none of the codeResponse